VstsClientLibrariesSamples/GettingStarted/Authentication.cs

```diff
@@ -1,26 +1,76 @@
-﻿using System;
-using System.Collections.Generic;
-
+using Microsoft.IdentityModel.Clients.ActiveDirectory;
 using Microsoft.TeamFoundation.Core.WebApi;
 using Microsoft.VisualStudio.Services.Common;
+using Microsoft.VisualStudio.Services.OAuth;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using AadAuthenticationContext = Microsoft.IdentityModel.Clients.ActiveDirectory.AuthenticationContext;
 
 namespace VstsClientLibrariesSamples.GettingStarted
 {
     public class Authentication
     {
-        IConfiguration _configuration;
+        // This is the hard coded Resource ID for Visual Studio Team Services, do not change this value
+        internal const string VSTSResourceId = "499b84ac-1321-427f-aa17-267ca6975798";
+
+        // This is the hard coded Resource ID for Graph, do not change this value
+        internal const string GraphResourceId = "https://graph.windows.net";
+
+        public Authentication()
+        {
+        }
 
-        public Authentication(IConfiguration configuration)
+        public IEnumerable<TeamProjectReference> InteractiveADAL(string vstsAccountName, string applicationId)
         {
-            _configuration = configuration;
+            AuthenticationContext authenticationContext = new AadAuthenticationContext("https://login.windows.net/common", validateAuthority: true);
+            var authenticationResultTask = authenticationContext.AcquireTokenAsync(VSTSResourceId, applicationId, new Uri("urn:ietf:wg:oauth:2.0:oob"), new PlatformParameters(PromptBehavior.Never)); 
+            AuthenticationResult authenticationResult = authenticationResultTask.Result;
+
+            VssOAuthAccessTokenCredential oAuthCredential = new VssOAuthAccessTokenCredential(authenticationResult.AccessToken);
+
+            return ListProjectsViaClientLibrary(vstsAccountName, oAuthCredential);
         }
 
-        public IEnumerable<TeamProjectReference> PersonalAccessToken(string url, string personalAccessToken)
+        public IEnumerable<TeamProjectReference> InteractiveADALExchangeGraphTokenForVSTSToken(string vstsAccountName, string applicationId)
+        {
+            AuthenticationContext authenticationContext = new AadAuthenticationContext("https://login.windows.net/common", validateAuthority: true);
+            var authenticationResultTask = authenticationContext.AcquireTokenAsync(GraphResourceId, applicationId, new Uri("urn:ietf:wg:oauth:2.0:oob"), new PlatformParameters(PromptBehavior.Never));
```

Comment:
You're using PromptBehavior.Never here. If this is the Interactive exchange method, that doesn't seem correct. Auto seems like a better choice.